### PR TITLE
ticket 5 completed

### PR DESCRIPTION
**Pull request by jhung2317**
added filter review by Category and show the single review page with all comments

### CODE REVIEW

File: src/components/CommentList.jsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }, []);
+
+  return (
+    <main>
+      <h4><u>Comments to This Review</u></h4>
+      {commentList.map((comment) => {
+          return <CommentCard key={comment.comment_id} comment={comment} />;
+        })}
+    </main>
+  );
```
Comment (Rayhaan2k9) [inline src/components/CommentList.jsx:59]: This works great on reviews that do have comments. However, when you go to a review that has no comments, the page breaks. For example review id 14. Would be great to add a message to the page that says it has no comments if this is the case

File: src/components/ReviewList.jsx [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
-function ReviewList({reviewList, setReviewList}) {
+function ReviewList({filter}) {
+  const [reviewList, setReviewList] = useState([]);
 
   useEffect(() => {
-    getAllReviews().then((review) => {
+    getAllReviews(filter).then((review) => {
       setReviewList(review)
     });
   }, []);
```
Comment (Rayhaan2k9) [inline src/components/ReviewList.jsx:15]: No loading state to let users know the reviews are loading. Would be great if you can add one :)

File: src/components/ReviewPage.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    comment_count,
+  } = singleReview;
+  const formattedDate = moment(created_at).format("D MMM YYYY");
+
+  useEffect(() => {
+    getReviewById(review_id)
+      .then((review) => {
+        setSingleReview(review[0])
+      })
+  }, []);
```
Comment (Rayhaan2k9) [inline src/components/ReviewPage.jsx:31]: No loading state for a single review page here as well. Please add one

File: src/utils/utils.js [hunk 4]
```diff
@@ -4,25 +4,45 @@ const API = axios.create({ baseURL: 'https://ncgames-tsyc.onrender.com/api' })
...
 
-export const getAllReviews = () => {
+export const getAllReviews = (filter) => {
   return API.get("/reviews")
-    .then(({data}) => {
-      console.log(data.Review)
+  .then(({data}) => {
+    if (filter !== ""){
+      const filteredArray = data.Review.filter((review) => review.category === filter)
+      return filteredArray
```
Comment (Rayhaan2k9) [inline src/utils/utils.js:20]: I see what you are doing here, however, this is a future ticket. Please delete this for now. Also, you will be able to achieve filtering by adding a query to your api call eg `/api/reviews?category=dexterity` If you reached ticket 11 in your backend, you will be able to do this. If you didn't, there is no need to implement the feature 🙂

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Great work on this Jason, however this is 2 tickets in 1 (5 and 6). Please ensure you are only working on the instructions from each ticket and sending a PR for each one separately. I've left a few comments for you. Please get these fixed and then resubmit this PR 🙂

File: src/utils/utils.js (hunk 4 above)
Comment (jhung2317) [reply src/utils/utils.js:20]: I see, I tried to fetch the data by API call as I heard someone mention in the seminar. But later I found that "/api/reviews?category=dexterity" doesn't give the data I want as I don't know I have to reach ticket 11 in my backend. This filter here is just for the user to get reviews via "All Category" in Nav bar. ( the logic is "all category" button sets the filter to blank while other categories set the filter to that specific name. If the filter is blank all reviews are shown)

File: src/components/CommentCard.jsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import moment from "moment";
+import './css/reviewPage.css'
+
+function CommentCard (comment) {
+  const { comment_id, votes, created_at, author, body } = comment.comment;
+
+  const formattedDate = moment(created_at).format("D MMM YYYY");
+  console.log(comment)
```
Comment (B7Bprog) [inline src/components/CommentCard.jsx]: Please remove all logs!

File: src/components/CommentList.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    getCommentsByReviewId(review_id)
+      .then((comment) => {
+        setCommentList(comment)
+        setIsLoading(false);
+      })
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading...</p>;
+  }
```
Comment (B7Bprog) [inline src/components/CommentList.jsx:22]: Great that you included the loading message for the comments.

File: src/components/CommentList.jsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+  if (!commentList) {
+    return (
+      <main >
+        <h4><u>Comments to This Review</u></h4>
+        <h3>
+          There is no comment for this review. Wanna start a conversation?
```
Comment (B7Bprog) [inline src/components/CommentList.jsx:28]: Nice refactor to add this message where there are no comments.

File: src/components/CommentList.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        </h3>
+        <form>
```
Comment (B7Bprog) [inline src/components/CommentList.jsx:30]: Not sure what the purpose of this input is as it doesn't do anything here.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Jason, please see my comments for the Ticket 6 part and refactor your code, then feel free to merge. 👍

File: src/components/CommentList.jsx (hunk 8 above)
Comment (jhung2317) [reply src/components/CommentList.jsx:30]: Yes it doesn't do anything. I just put an input box here for adding new comment which I am going to do later

**Pull request merged**